pinot-server/src/test/java/org/apache/pinot/server/api/BaseResourceTest.java

```diff
@@ -58,8 +59,8 @@
   protected static final String TABLE_NAME = "testTable";
 
   private final Map<String, TableDataManager> _tableDataManagerMap = new HashMap<>();
-  protected final List<ImmutableSegment> _indexSegments = new ArrayList<>();
-
+  protected final List<ImmutableSegment> _realtimeIndexSegments = new ArrayList<>();
```

Comment:
Why do you need to separate realtime and offline segments? They should be handled in the same way right?
Also the realtime index segments are not CONSUMING segment, but COMPLETED segment, which is a little bit confusing.

Reply:
Before we talk about the separation of two kinds of segments, I want to point out this test was set up almost like a mini integration tests with mock servers and download clients. So ideally it should reflect a true Pinot server setup environment with proper directory naming associated with table name and type -- this was NOT done earlier i think (e.g., the test table name does not have type). The modified test code adheres to these expectations. The separation of segments also follows the same idea: they are just different segments.

For your comments about realtime index segment, I believe you are referring to the fact that CONSUMING segment could not be downloaded just like COMPLETED segment, right? Yes, that is true for now and we can add a note to API as well.



.

Reply:
I agree the test setup should reflect the true cluster setup, where the table name should have the correct type suffix.
My concern about the CONSUMING segment is because the CONSUMING segment is not ImmutableSegment, so I'm not sure if adding separate realtime segments as a list of ImmutableSegment is the correct design.
Currently the test setup is mimicking a offline only table, so I would suggest introducing the realtime part as a separate PR (the fix for the table name is good).

Reply:
To me, this new API and its test provides downloading of immutable (or non-consuming) segments. So I agree there is a subset of realtime segments not covered here. We can either rename the API or add notes for this important subtlety.  Given this API is so new and our main design is to download immutable segments, I added notes to the API to highlight this point.

Since the code still allows immutable realtime segment downloads, I feel the test should also cover this case.